sentry_sdk/integrations/celery.py

```diff
@@ -37,6 +42,23 @@ def sentry_build_tracer(name, task, *args, **kwargs):
         ignore_logger("celery.worker.job")
 
 
+def _wrap_apply_async(task, f):
+    def apply_async(self, *args, **kwargs):
+        hub = Hub.current
+        integration = hub.get_integration(CeleryIntegration)
+        if integration is not None and integration.propagate_traces:
+            headers = None
+            for key, value in hub.iter_trace_propagation_headers():
+                if headers is None:
+                    headers = dict(kwargs.get("headers") or {})
+                headers[key] = value
+            if headers is not None:
+                kwargs["headers"] = headers
```

Comment:
Can you do that? Won't those kwargs not be visible to the task?

Reply:
Those are the `kwargs` to the `delay` function which takes those arguments. I tried first to use the signals but the schedule signal is only implemented for amqp but not eager mode which is used for the unittests.